src/main/java/gyro/google/cloudsql/DbSettings.java

```diff
@@ -0,0 +1,727 @@
+            config.copyFrom(model.getBackupConfiguration());
+            setBackupConfiguration(config);
+        }
+
+        setDataCacheConfig(null);
+        if (model.getDataCacheConfig() != null) {
+            DbDataCacheConfig config = newSubresource(DbDataCacheConfig.class);
+            config.copyFrom(model.getDataCacheConfig());
+            setDataCacheConfig(config);
+        }
+
+        setInsightsConfig(null);
+        if (model.getInsightsConfig() != null) {
+            DbInsightsConfig config = newSubresource(DbInsightsConfig.class);
+            config.copyFrom(model.getInsightsConfig());
+            setInsightsConfig(config);
+        }
+
+        setIpConfiguration(null);
+        if (model.getIpConfiguration() != null) {
+            DbIpConfiguration config = newSubresource(DbIpConfiguration.class);
+            config.copyFrom(model.getIpConfiguration());
+            setIpConfiguration(config);
+        }
+
+        setLocationPreference(null);
+        if (model.getLocationPreference() != null) {
+            DbLocationPreference config = newSubresource(DbLocationPreference.class);
+            config.copyFrom(model.getLocationPreference());
+            setLocationPreference(config);
+        }
+
+        setMaintenanceWindow(null);
+        if (model.getMaintenanceWindow() != null) {
+            DbMaintenanceWindow config = newSubresource(DbMaintenanceWindow.class);
+            config.copyFrom(model.getMaintenanceWindow());
+            setMaintenanceWindow(config);
+        }
+
+        setPasswordValidationPolicy(null);
+        if (model.getPasswordValidationPolicy() != null) {
+            DbPasswordValidationPolicy config = newSubresource(DbPasswordValidationPolicy.class);
+            config.copyFrom(model.getPasswordValidationPolicy());
+            setPasswordValidationPolicy(config);
+        }
+
+        getDatabaseFlags().clear();
+        if (model.getDatabaseFlags() != null) {
+            for (DatabaseFlags flag : model.getDatabaseFlags()) {
+                DbDatabaseFlag databaseFlag = newSubresource(DbDatabaseFlag.class);
+                databaseFlag.copyFrom(flag);
+                getDatabaseFlags().add(databaseFlag);
+            }
+        }
+
+        getDenyMaintenancePeriods().clear();
+        if (model.getDenyMaintenancePeriods() != null) {
+            for (DenyMaintenancePeriod period : model.getDenyMaintenancePeriods()) {
+                DbDenyMaintenancePeriod denyMaintenancePeriod = newSubresource(DbDenyMaintenancePeriod.class);
+                denyMaintenancePeriod.copyFrom(period);
+                getDenyMaintenancePeriods().add(denyMaintenancePeriod);
+            }
+        }
+    }
+
+    @Override
+    public String primaryKey() {
+        return "";
+    }
+
+    public Settings toSettings() {
+        Settings settings = new Settings();
+        settings.setSettingsVersion(getSettingsVersion());
```

Comment:
This will probably be a required field then.

Also given this is a db settings, re check to see if there are other fields that are required.

Reply:
The settings version is an output field so I don't think it's need the annotation.
It just needs to be provided on every update call.

The tier is basically the only thing required other than the name.